object_gomap_test.go

```diff
@@ -326,3 +328,154 @@ func TestGoMapUnicode(t *testing.T) {
 		t.Fatalf("Unexpected value: %v", res)
 	}
 }
+
+func TestGoMapMemUsage(t *testing.T) {
+	vm := New()
+	vmCtx := NewMemUsageContext(vm, 100, 100, 100, 100, nil)
+
+	nestedMap := map[string]interface{}{
+		"subTest1": valueInt(99),
+		"subTest2": valueInt(99),
+		"subTest3": valueInt(99),
+		"subTest4": valueInt(99),
+	}
+
+	// the baseObject is quite large when ToValue is called due to the functions on the object
+	// calculating ahead of time for test case
+	nestedMapAsObject := vm.ToValue(nestedMap)
+	nestedMapMemUsage, nestedMapNewMemUsage, err := nestedMapAsObject.MemUsage(vmCtx)
+	if err != nil {
+		t.Fatalf("Unexpected error. Actual: %v Expected: %v", err, nil)
+	}
+
+	tests := []struct {
+		name           string
+		val            *objectGoMapSimple
+		threshold      int
+		expectedMem    uint64
+		expectedNewMem uint64
+		errExpected    error
+	}{
+		{
+			name:      "should account for each key value pair given a non-empty object",
+			threshold: 100,
+			val: &objectGoMapSimple{
+				baseObject: baseObject{
+					val: &Object{runtime: vm},
+				},
+				data: map[string]interface{}{
+					"test0": valueInt(99),
+					"test1": valueInt(99),
+					"test2": valueInt(99),
+					"test3": valueInt(99),
+				},
+			},
+			// overhead + len("testN") + value
+			expectedMem: SizeEmptyStruct + (5+SizeInt)*4,
+			// overhead + len("testN") with string overhead + value
+			expectedNewMem: SizeEmptyStruct + ((5+SizeString)+SizeInt)*4,
+			errExpected:    nil,
+		},
+		{
+			name:      "should account for each key value pair given map with a nil value",
+			threshold: 100,
+			val: &objectGoMapSimple{
+				baseObject: baseObject{
+					val: &Object{runtime: vm},
+				},
+				data: map[string]interface{}{
+					"test": nil,
+				},
+			},
+			// overhead + len("test") + null
+			expectedMem: SizeEmptyStruct + 4 + SizeEmptyStruct,
+			// overhead + len("test") with string overhead + null
+			expectedNewMem: SizeEmptyStruct + (4 + SizeString) + SizeEmptyStruct,
+			errExpected:    nil,
+		},
+		{
+			name:      "should account for nested reflect object",
+			threshold: 100,
+			val: &objectGoMapSimple{
+				baseObject: baseObject{
+					val: &Object{runtime: vm},
+				},
+				data: map[string]interface{}{
+					"test": &objectGoMapSimple{
+						baseObject: baseObject{
+							val: &Object{runtime: vm},
+						},
+						data: map[string]interface{}{
+							"subTest1": valueInt(99),
+							"subTest2": valueInt(99),
+							"subTest3": valueInt(99),
+							"subTest4": valueInt(99),
+						},
+					},
+				},
+			},
+			// overhead + len("test") + reflectObject
+			expectedMem: SizeEmptyStruct + 4 + SizeEmptyStruct,
+			// overhead + len("test") with string overhead + reflectObject
+			expectedNewMem: SizeEmptyStruct + 4 + SizeString + SizeEmptyStruct,
+			errExpected:    nil,
+		},
```

Comment:
[minor] Kind of like I mentioned in the answers to your questions I would convert this to have `"test": &nestedMap`